horace_core/sqw/@sqw/section.m

```diff
@@ -0,0 +1,159 @@
+function wout = section (win,varargin)
+% Takes a section out of an sqw object
+%
+%   >> wout = section (win, [ax_1_lo, ax_1_hi], [ax_2_lo, ax_2_hi], ...)
+%
+% Input:
+% ------
+%   win                 Input sqw object
+%
+%   [ax_1_lo, ax_1_hi]  Lower and upper limits for the first axis. Bins are retained whose
+%                      centres lie in this range.
+%                       To retain the limits of the input structure, type '', [], or the scalar '0'
+%
+%   [ax_2_lo, ax_2_hi]  Lower and upper limits for the second axis
+%
+%           :                       :
+%
+%       for as many axes as there are plot axes
+%
+% Output:
+% -------
+%   wout                Output dataset.
+%
+%
+% Example: to alter the limits of the first and third axes of a 3D sqw object:
+%   >> wout = section (win, [1.9,2.1], [], [-0.55,-0.45])
+
+
+% Original author: T.G.Perring
+%
+
+
+% Trivial case of no section arguments
+if nargin==1
+    wout = copy(win);
+    return
+end
+
+% Dimension of input data structures
+ndim=dimensions(win(1));
+for i=2:numel(win)
+    if dimensions(win(i))~=ndim
+        error('All objects must have same dimensionality for sectioning to work')
+    end
+end
+if ndim==0  % no sectioning possible
+    error ('Cannot section a zero dimensional object')
+end
+
+
+nargs= length(varargin);
+if nargs~=ndim
+    error ('Check number of arguments')
+end
+
+% Initialise output argument
+wout = copy(win);
+
+tol=4*eps('single');    % acceptable tolerance: bin centres deemed contained in new boundaries
+
+for n=1:numel(win)
+    [ndim,sz]=dimensions(win(n));   % need to get sz array specific for each element in array win
+    % Get section parameters and axis arrays:
+    % The input sectioning arguments refer to the *display* axes; these must be converted to the relevant plot axes in the algorithm
+    irange = zeros(2,ndim);
+    array_section = cell(1,ndim);
+    p=win(n).data.p;   % extract bin boundaries
+    for i=1:nargs
+        if isempty(varargin{i}) || (isscalar(varargin{i}) && isequal(varargin{i},0))
```

Comment:
what is `varargin{1}`

Reply:
It says varargin{i} not varargin{1} - it is the relevant [lo,hi] as described in the args after win

Reply:
ok, this raw varargin handling is slightly opaque but will be resolved as part of #578